stubs/python-decouple/decouple.pyi

```diff
@@ -0,0 +1,168 @@
+from _typeshed import Incomplete
+from collections import OrderedDict
+from collections.abc import Callable, Iterable, Sequence
+from configparser import ConfigParser
+from typing import Any, Generic, Protocol, TextIO, TypeVar, overload
+
+_T = TypeVar("_T")
+_TCsv = TypeVar("_TCsv")
+
+PYVERSION: Incomplete  # undocumented
+DEFAULT_ENCODING: str  # undocumented
+TRUE_VALUES: set[str]  # undocumented
+FALSE_VALUES: set[str]  # undocumented
+
+def read_config(parser: ConfigParser, file: TextIO) -> None: ...  # undocumented
+def strtobool(value: str | bool) -> bool: ...  # undocumented
+
+class UndefinedValueError(Exception): ...
+class Undefined: ...
+
+undefined: Undefined
+
+class Config:
+    repository: _Repository
+    def __init__(self, repository: _Repository) -> None: ...
+    @overload
+    def get(self, option: str, default: str = ...) -> str: ...
+    @overload
+    def get(self, option: str, default: _T) -> str | _T: ...
+    @overload
+    def get(self, option: str, *, cast: Callable[[str], _T]) -> _T: ...
+    @overload
+    def get(self, option: str, default: str | _T, cast: Callable[[str], _T]) -> _T: ...
+    @overload
+    def __call__(self, option: str, default: str = ...) -> str: ...
+    @overload
+    def __call__(self, option: str, default: _T) -> str | _T: ...
+    @overload
+    def __call__(self, option: str, *, cast: Callable[[str], _T]) -> _T: ...
+    @overload
+    def __call__(self, option: str, default: str | _T, cast: Callable[[str], _T]) -> _T: ...
+
+class _Repository(Protocol):  # undocumented
+    def __contains__(self, key: str) -> bool: ...
+    def __getitem__(self, key: str) -> str: ...
+
+class RepositoryEmpty:  # undocumented
+    def __init__(self, source: str = ..., encoding: str = ...) -> None: ...
+    def __contains__(self, key: str) -> bool: ...
+    def __getitem__(self, key: str) -> str: ...
```

Comment:
This actually returns None. That's inconsistent with the protocol though, and maybe reflecting it accurately would make the types too complicated. Seems worth a comment, though.

Reply:
Yes, this is a complicated situation. I added a comment.